### PR TITLE
Add Arrow scanning for fixed size list

### PR DESCRIPTION
**Pull request by manh9203**
This PR resolved the issue #3237.

### CODE REVIEW

File: tools/python_api/test/test_df_pyarrow.py [hunk 1]
```diff
@@ -259,6 +259,32 @@ def test_pyarrow_list(conn_db_readonly : ConnDB) -> None:
     
     assert idx == len(index)
 
+def test_pyarrow_fixed_list(conn_db_readonly : ConnDB) -> None:
```
Comment (andyfengHKU) [inline tools/python_api/test/test_df_pyarrow.py:262]: can u test fixed list with different types.

**Review by andyfengHKU — APPROVED** [review]
Overall LGTM can @mxwli take another look here.

**Comment by mxwli** [comment]
Looks good :+1:

**Comment by mxwli** [comment]
While reading this code, I found a part of the code that I had overlooked previously. I've applied fixes to other datatypes (will be in a different PR), and am pushing a fix specific to fixed-list to this branch.

**Review by andyfengHKU — APPROVED** [review]

**Pull request merged**